### ISSUE TITLE
Unable to compile in 4.1.3

### ISSUE THREAD

**Issue by kwik-silver** [shin-gamedev/gbs-plugins#3]
When I "run game" with the plugin installed to the correct folder, in GB Studio 4.1.3, the game does not compile. The following error is returned. 

```
?ASlink-Warning-Undefined Global '_swimming' referenced by module ''
  
?ASlink-Warning-Undefined Global '_solid_ladders' referenced by module ''
```
 
This also occurs after emptying the build cache. 

The same game compiles fine w/o this plugin.

**Comment by kwik-silver** [comment]
This is with both the 4.0.0 and 4.1.2 versions.

**Comment by shin-gamedev** [comment]
Which version of Adventure Reforged are you using? There shouldn't be any compatibility issues, but I can check. Try reinstalling from scratch in the meantime.

**Comment by kwik-silver** [comment]
This is on the 4.1.2 version of the plugin. I tried starting a brand new project and added the plugin to that. In this case, it worked fine! 

However, I am trying to add this to an existing project. Is that supported?

**Comment by kwik-silver** [comment]
Despite GB Studio saying I am on version 4.1.3 of the application, a warning I get (from another plugin, specifically [this one](https://github.com/Mico27/GBS-scrollScenePlugin)) read:

`Plugin "GBS-scrollScenePlugin-main" was created for an older version of GB Studio and may not work correctly. Plugin uses engine "4.0.0-e0" but GB Studio is using engine "4.2.0-e1". `

Not sure why these numbers are different, but including this info for you if relevant. I am also only using one other plugin, the [Draw Lifebar listed here](https://github.com/pau-tomas/gb-studio-plugins?tab=readme-ov-file). I can't imagine either of these would impact your plugin.

**Comment by kwik-silver** [comment]
Actually, upon further testing this seems to precisely be an incompatibility with issue with GBS-scrollScenePlugin-main. I added that plugin to the brand new test project, and I got the exact same error. This plugin also has a file, "adventure.c". 

I would say "issue closed", since this is really an incompatibility issue, but I leave that up to you. I'm going to see if I can hack these two plugins to play nice together.
